fix: Update cert-manager to use crds.enabled instead of the deprecated installCRDs option

kinyat opened the pull request:
### What does this PR do?

This is a follow-up to https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/pull/435.  

<strike>Instead of changing the option directly, we add a pre-check before setting `installCRDs`.</strike>

<strike>If the user has already set the new `crds.enabled` flag in either `values` or `set`, we do not add the deprecated `installCRDs` option.  </strike>

<strike>This helps prevent breaking changes and allows users to avoid the deprecated installCRDs option.</strike>

Update: Follow-up to https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/pull/443#pullrequestreview-2633785239

I have updated the PR to pump up the version of the cert-manager to the latest minor version.

I have also updated the PR to remove the usage of `installCRDs` and replaced it with both the new `crds.enabled` and `crds.keep` options.

### Motivation
- Resolves https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/issues/420

### More

- [ ] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I ran `pre-commit run -a` with this PR

### For Moderators

- [ ] E2E Test successfully complete before merge?

### Additional Notes

<!-- Anything else we should know when reviewing? -->

askulkarni2 reviewed (commented):
Lets just bump up the version of cert-manager and set `crds.enabled` instead of `installCRDs`. We can break compatibility with older versions because we use semantic versioning on the module.

kinyat commented:
> Lets just bump up the version of cert-manager and set `crds.enabled` instead of `installCRDs`. We can break compatibility with older versions because we use semantic versioning on the module.

I have updated the PR to pump up the version of the cert-manager to the latest minor version.
I have also updated the PR to remove the usage of `installCRDs` and replaced it with both the new `crds.enabled` and `crds.keep` options.

Can you please give it a 2nd look?

hgadham commented on main.tf:
```diff
@@ -1911,7 +1911,11 @@ module "cert_manager" {
+      value = true
+    },
+    {
+      name  = "crds.keep"
```
I don't recommend hardcoding this. I would suggest to remove this entirely and let helm choose these values

hgadham replied:
Additionally, I should have the flexibility to choose whether to manage CRDs through the addon or deploy them externally.

hgadham replied:
Hardcoding limits my option here.

hgadham replied:
If it is preferred option, why are we not consistently doing it for other addons.

kinyat commented:
> > Lets just bump up the version of cert-manager and set `crds.enabled` instead of `installCRDs`. We can break compatibility with older versions because we use semantic versioning on the module.
> 
> I have updated the PR to pump up the version of the cert-manager to the latest minor version. I have also updated the PR to remove the usage of `installCRDs` and replaced it with both the new `crds.enabled` and `crds.keep` options.
> 
> Can you please give it a 2nd look?

@askulkarni2 Hi 👋 Just want to ping you again for a 2nd review 🙏 Thanks

kinyat commented:
@askulkarni2 Hi 👋 Just want to ping you again for a 2nd review 🙏 Thanks
